src/staging.c

```diff
@@ -148,8 +148,8 @@ int do_staging(struct file *file, struct manifest *MoM)
 			ret = -errno;
 			goto out;
 		}
-		string_or_die(&tarcommand, "tar -C %s " TAR_PERM_ATTR_ARGS " -cf - '%s' 2> /dev/null | "
-					   "tar -C %s%s " TAR_PERM_ATTR_ARGS " -xf - 2> /dev/null",
+		string_or_die(&tarcommand, TAR_COMMAND " -C %s " TAR_PERM_ATTR_ARGS " -cf - %s 2> /dev/null | "
```

Comment:
ok, we're suffering from quotes and other kind of injections possible with `system()`. My recommendation is for you guys to do a review and replace these with non-`system()` variants where you would avoid calling `sh -c "string"` that is implied from `system()`, using a `fork()` + `execv()` + `waitpid()`